### ISSUE TITLE
bug in angular parser/formatter - puts brackets in wrong place which breaks the logic

### ISSUE THREAD

**Issue by kirillgroshkov** [prettier/prettier#5475]
Prettier puts brackets in such place that breaks the angular pipe logic.

**Prettier 1.15.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEACVAeAhgBxwWgGcBXAIwFsBLGfU4mGaAHSnTdTABstDCBeJiABOcAOaVCMIVhiVoAYQgATOACU4JTjAD62wpVFRiOAEL1GUQS3boA2qJEIAugPDQpPGFdY370pZQIMC6CAISQsNKS3jaotgGEWKSccADiABYQkiEgUsRwMTYAfMDA1rGoABRgWEIwqAA+qDwAnlBgAJQAdIwwWJwACkKUYHBdWOQQxLCoAPyoAOSQKvhYYKM4MAuoSIvLcKvrcJv4AGaOC+UVjagw5Eq71bX1Ta3tHVcVTZLDOAAqWFEhE+6AAvqCMAB6XAEEgUai0czMHzoIogAA0IAgmzkUEIyFAtSEEAA7gNagh8SgsAA3CCUJQYkCkaRgADWcBgAGUcGtKFBRMg8nBMek7pwAOrpagaXmjLmU6iUGnUFrIcC8Jn8whwOpDQHkLDIU79HWYgBWhAAHiZWRzuRM4AAZflwY2mkUgS1Wrn80QpACKxAg8HdnDNIF5Qh1QnVWAFxG4scxOGGsAlDJg6WQAA4AAwp4k6iXSHDq1MaXU0t2YkQAR2IlBE+tEhrDEZ1VCFQnymP0AsDwdDSBN4c9fVIGaUWeQACZMR5KJw-YpyG2UFBoDWQMQdQDSFTRzrwUA)
```sh
--parser angular
```

**Input:**
```html
  <app-submit-button
      class="registrationCodeResult__signupButton"
      >{{
        (cart | async).totalPrice.amount ? 'code-accept' : 'code-accept-free'
          | tmd: (cart | async)
          | stripTags
      }}</app-submit-button
    >
```

**Output:**
```html
<app-submit-button
  class="registrationCodeResult__signupButton"
  >{{
    (cart | async).totalPrice.amount
      ? "code-accept"
      : ("code-accept-free" | tmd: (cart | async) | stripTags)
  }}</app-submit-button
>

```

**Expected behavior:**

```html
<app-submit-button
  class="registrationCodeResult__signupButton"
  >{{
    (cart | async).totalPrice.amount
      ? "code-accept"
      : "code-accept-free" | tmd: (cart | async) | stripTags
  }}</app-submit-button
>

```

**Comment by ikatyang** [comment]
I tried it on the official Angular parser (`@angular/compiler/src/expression_parser`) and they look the same:

![image](https://user-images.githubusercontent.com/8341033/48472010-3b8d7280-e830-11e8-87a7-5c6a2cf939e4.png)

![image](https://user-images.githubusercontent.com/8341033/48472050-565fe700-e830-11e8-9938-f857ea948b87.png)

**Comment by TocabotJames** [comment]
welp, actually looks like thats already fixed. needed to restart vscode to get it to use the latest prettier

**Comment by ikatyang** [comment]
@kirillgroshkov Since it's the official one, its behavior should be exactly the same as you compiled. The only possible issue I can think of is the version, I got the same result on both Angular 6 and 7, what version of Angular are you using?

**Comment by ikatyang** [comment]
@TocabotJames The issue you mentioned should be fixed by #5387, which is available in 1.15.2.

(I'm going to hide your comment to keep this thread clear.)

**Comment by kirillgroshkov** [comment]
@ikatyang We're using Angular 7 there.

```
yarn list @angular/*
├─ @angular/animations@7.0.3
├─ @angular/cli@7.0.5
├─ @angular/common@7.0.3
├─ @angular/compiler-cli@7.0.3
├─ @angular/compiler@7.0.3
├─ @angular/core@7.0.3
├─ @angular/forms@7.0.3
├─ @angular/language-service@7.0.3
├─ @angular/platform-browser-dynamic@7.0.3
├─ @angular/platform-browser@7.0.3
└─ @angular/router@7.0.3
```

But. I'm just trying to understand your line of thought. We can both see that issue is reproducible in Prettier playground that shows `1.15.2`. I don't know which version of Angular is used there (in the playground). But it produces the same result as for us locally.

**Comment by ikatyang** [comment]
We use the parser from Angular 6, but there should be no difference given that there's only [a typo fix commit](https://github.com/angular/angular/commits/7.0.x/packages/compiler/src/expression_parser) between Angular 6.1.10 (Oct 11) and 7.0.3 (Nov 8), and also I've tested it locally using Angular 7.

Would you mind to run this code in your project? If the output is `Conditional` (this is the expected output, same as ours) then it must be something wrong in the compiling instead of the parsing. If the output is `BindingPipe`, ...I can't think of a reason how it can happen.

```js
const ast = require("@angular/compiler").parseTemplate(`{{

    (cart | async).totalPrice.amount ? 'code-accept' : 'code-accept-free'
          | tmd: (cart | async)
          | stripTags

}}`)

console.log(ast.nodes[0].value.ast.expressions[0].constructor.name)
```

**Comment by kirillgroshkov** [comment]
Ok, finally I understand, sorry.

I tried it here and actually confirmed that you were right - that behaviour of prettier doesn't change functionality in Angular. I was wrong. 

https://stackblitz.com/edit/angular-playground?file=app%2Fapp.component.html

I expected that this `name === 'Angular' ? 'ang' : 'not_ang' | uppercase }}` would uppercase the string in both cases of ternary. But it only uppercases in second case.

The root issue we had was actually what was fixed there: #5387 #5397

And I confirm that is not reproducible any more in 1.15.2 (but did exist in 1.15.1).

Closing.

Thanks a lot for clarification!

Too bad it made a lot of noise for us internally because of this breakage.
